src/backend/utils/misc/guc.c

```diff
@@ -6155,6 +6156,16 @@ static struct config_int ConfigureNamesInt[] =
 	},
 
 	{
+		{"optimizer_insert_sort_partition_number", PGC_USERSET, DEVELOPER_OPTIONS,
+			gettext_noop("Minimum number of partitions required for sorting tuples during insertion in an append only row-oriented partitioned table"),
+			NULL,
+			GUC_NOT_IN_SAMPLE
+		},
+		&optimizer_insert_sort_partition_number,
+		INT_MAX, 0, INT_MAX, NULL, NULL
+	},
+
+	{
```

Comment:
Can you guys come up with a better GUC name for this?
@vraghavan78 @xinzweb @oarap